File: src/Java/Behavioral/Strategy/DispatchStrategy/EmailStrategy.java
package Java.Behavioral.Strategy.DispatchStrategy;

import Java.Behavioral.Strategy.Message;

/**
 * Created by Jack F. Dalton on 0029 29 08 2016.
 */
public class EmailStrategy implements DispatchStrategy {
    @Override
    public void send(Message message) {
        System.out.println("Message sent to " + message.getReceivingUserDetails().getFullName() + " via email");
        System.out.println("Content: " + message.getContent());
        System.out.println("Send Timestamp: " + message.getDateTimeSent() + "\n");
    }
}

File: src/Kotlin/strategyPattern/DispatchStrategies/EmailStrategy.kt
package Kotlin.strategyPattern.DispatchStrategies

import Kotlin.strategyPattern.Message

class EmailStrategy : DispatchStrategy {

    override fun send(message: Message) {
        println("Message sent to " + message.receivingUser.fullName + "Via email")
        println("Content: " + message.content)
        println("Send Timestamp: " + message.dateTimeSent + "\n")
    }
}
